contracts/utils/LibBytes.sol

```diff
@@ -128,4 +128,65 @@ library LibBytes {
     }
     return result;
   }
+
+  /**
+   * @dev Reads an uint16 value from a position in a byte array.
+   * @param data Byte array to be read.
+   * @param index Index in byte array of uint16 value.
+   * @return a uint16 value of data at given index.
+   * @return newIndex Updated index after reading the value.
+   */
+  function readUint16(
+    bytes memory data,
+    uint256 index
+  ) internal pure returns (uint16 a, uint256 newIndex) {
+    assembly {
+      let word := mload(add(index, add(32, data)))
+      a := and(shr(240, word), 0xffff)
+      newIndex := add(index, 2)
```

Comment:
What happens if you overflow? Require below would pass

Reply:
newIndex overflows, but because we only call this method in such a way that index is modified by static values (all below 256 bits), it's not possible go build a test case for it